#!/usr/bin/env python
PACKAGE = "cell_culture"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

pipette_status_enum = gen.enum([ gen.const("no_tip",      int_t, 0, "Pipette has no tip attached"),
                        gen.const("empty",      int_t, 1, "Pipette tip is empty"),
                        gen.const("media",     int_t, 2, "Pipette tip has media"),
                        gen.const("yeast",      int_t, 3, "Pipette tip has yeast"),
                        gen.const("contaminated",      int_t, 4, "Pipette tip is contaminated (empty) and needs to be removed"),],
                        "Pipette Status")

gen.add("tip_rack_count",    int_t,    0, "", 12,  0, 12)
gen.add("monitoring_idx",    int_t,    0, "", 0,  0, 7)
gen.add("holding_pipette",   bool_t,   0, "",  False)
# gen.add("pipette_has_tip",   bool_t,   0, "",  False)
gen.add("pipette_status", int_t, 0, "Set pipette status", 0, 0, 4, edit_method=pipette_status_enum)
gen.add("pipette_actuator_pos",   int_t, 0, "", 1850, 1300, 1850)
gen.add("needs_split",   str_t,   0, "",  "")
gen.add("needs_media",   str_t,   0, "",  "")
gen.add("needs_yeast",   str_t,   0, "",  "")
gen.add("shaker_active",   bool_t,   0, "",  False)

exit(gen.generate(PACKAGE, "cell_culture", "CellCultureGlobal"))